src/rules/noMagicNumbersRule.ts

```diff
@@ -62,15 +89,55 @@ export class Rule extends Lint.Rules.AbstractRule {
 
     public static DEFAULT_ALLOWED = [ -1, 0, 1 ];
 
+    /* tslint:disable no-unsafe-any */
     public apply(sourceFile: ts.SourceFile): Lint.RuleFailure[] {
-        const allowedNumbers = this.ruleArguments.length > 0 ? this.ruleArguments : Rule.DEFAULT_ALLOWED;
-        return this.applyWithWalker(new NoMagicNumbersWalker(sourceFile, this.ruleName, new Set(allowedNumbers.map(String))));
+        return this.applyWithWalker(
+            new NoMagicNumbersWalker(
+                sourceFile, this.ruleName,
+                this.parseOptions(this.ruleArguments[0]),
+            ),
+        );
+    }
+
+    private parseOptions(args: any): Options {
+        /* For legacy config support */
+        if (args !== undefined && args.length && typeof args[0] === "number") {
+            return {
+                allowElementAccess: false,
+                allowedNumbers: new Set((args as number[]).map(String)),
+            };
+        }
+
+        const options: Options =
+            args !== undefined
+                ? {
+                    allowElementAccess: (args as {[key: string]: boolean})["allow-element-access"],
+                    /* tslint:disable-next-line strict-boolean-expressions */
+                    allowedNumbers: args["allowed-numbers"] || Rule.DEFAULT_ALLOWED,
+                }
+                : {
+                    allowElementAccess: false,
+                    allowedNumbers: Rule.DEFAULT_ALLOWED,
+                };
+
+        options.allowedNumbers =
+            new Set((options.allowedNumbers as number[]).map(String));
+
+        return options;
     }
+    /* tslint:enable no-unsafe-any */
 }
 
-class NoMagicNumbersWalker extends Lint.AbstractWalker<Set<string>> {
+class NoMagicNumbersWalker extends Lint.AbstractWalker<Options> {
     public walk(sourceFile: ts.SourceFile) {
         const cb = (node: ts.Node): void => {
+            if (
+                commentPositions(this.sourceFile)
```

Comment:
❌ strongly against this feature. instead of implicitly disabling the rule if a comment appears, users should explicitly use the **existing `// tslint:disable-next-line` feature**, which is always a comment itself!

Reply:
I see a big difference here:

```ts
foo = bar[1]; // some helpful info about "1"
```

and
```ts
// tslint:disable-next-line no-magic-numbers
foo = bar[1];
```

Please keep in mind that this PR is in response to issues about the strictness of no-magic-numbers.